src/ERC721Drop.sol

```diff
@@ -1267,6 +1272,38 @@ contract ERC721Drop is
         emit MintFeePayout(zoraFee, ZORA_MINT_FEE_RECIPIENT, success);
     }
 
+    function _requireCanMintQuantity(uint256 quantity) internal view {
+        if (quantity + _totalMinted() > config.editionSize) {
+            revert Mint_SoldOut();
+        }
+    }
+
+    function _handleSupplyRoyalty(uint256 mintAmount) internal returns (uint256 totalRoyaltyMints) {
+        if (royaltyMintSchedule == 0) {
+            // If we still have no schedule, return 0 supply royalty.
+            return 0;
+        }
+
+        totalRoyaltyMints = (mintAmount + (_totalMinted() % royaltyMintSchedule)) / (royaltyMintSchedule - 1);
+
+        if (totalRoyaltyMints > 0) {
+            address royaltyRecipient = config.fundsRecipient;
+
+            // If we have no recipient set, return 0 supply royalty.
+            if (royaltyRecipient == address(0)) {
+                return 0;
+            }
+            _mintNFTs(royaltyRecipient, totalRoyaltyMints);
+        }
+    }
+
+    function updateRoyaltyMintSchedule(uint32 newSchedule) external onlyAdmin {
```

Comment:
There would be some issues that happen if this is updated after a few tokens have been minted already:

For example:
* the royalty mint schedule is 5
* 4 tokens are minted
* the royalty schedule is changed to 4
* a royalty would only be received when the mint count is 8

We should either:
* Not allow this to be updated after the first token is minted
* Store how many royalty mints there have been done so far, and compare that to how many there should be at this point, and mint the difference

Reply:
great point @oveddan , main goal was to get parity with 1155 which has this same impl but lets talk about this today

Reply:
@oveddan I think missing during update is expected – it's every Nth from that point forward